### ISSUE TITLE
Possible bug with the DQFD agent (v0.3.0)

### ISSUE THREAD

**Issue by Mahgoobi** [reinforceio/tensorforce#208]
First of all, thanks for such a great library. 
After the new update (0.3.0) I tried to test the DQFD agent with the new configuration structure. However, I am getting some error and I think its because of a bug in the code.

My test looks like the following:

```python
from tensorforce import Configuration
from tensorforce.agents import DQFDAgent

network_spec = [
    dict(type = 'dense', size = 50),
    dict(type = 'dense', size = 50),
    dict(type = 'dense', size = 50)
]

states_spec = dict(shape = (10,), type = "float")
actions_spec = dict(continuous = False, num_actions = 2)

config = Configuration(
    memory = dict(
        type = 'replay',
        capacity = 1000
    ),
    batch_size = 32,
    first_update = 10,
    target_sync_frequency = 10,
    demo_memory_capacity = 100000,
    demo_sampling_ratio = 0.2,
    target_update_frequency = 10
)

agent = DQFDAgent(
    states_spec = states_spec,
    actions_spec = actions_spec,
    network_spec = network_spec,
    config = config
)
```

This code throws the following error:

```shell
File "usr/local/lib/python2.7/dist-packages/tensorforce/agents/dqfd_agent.py", line 132, in __init__
self.demo_memory = Replay(self.demo_memory_capacity, self.states_spec, self.actions_spec)
AttributeError: `DQFDAgent` object has no attribute 'state_spec`
```

Which didn't make since to me since I am actually passing states_spec to the object. 
I did some traceback and I found [this part of the code](https://github.com/reinforceio/tensorforce/blob/046954c1bf43d3464e3d719ce8b074f9a9453882/tensorforce/agents/dqfd_agent.py#L141-L150) which where the problem might be. 

state_spec is read before it is initialized? Or am I required to explicitly create a MemoryAgent before I create a DQFDAgent?
```python
self.demo_memory = Replay(self.demo_memory_capacity, self.states_spec, self.actions_spec)
```

**Comment by michaelschaarschmidt** [comment]
Thanks for raising, yeah states spec is initialized in the super call, let me just check why this didnt show up in a test

**Comment by michaelschaarschmidt** [comment]
Looks like 02fb71f730f6e600edbf27f58cea86209cb5dc55 broke it.

**Comment by michaelschaarschmidt** [comment]
Fixed via https://github.com/reinforceio/tensorforce/commit/5e523a84be8ae34fcc86b45deda36d2148acd673

**Comment by Mahgoobi** [comment]
Thanks @michaelschaarschmidt for the fix. 
Are you guys releasing a newer version (perhaps 0.3.1) anytime soon?

**Comment by michaelschaarschmidt** [comment]
Yes, probably a few more bug-fixes that will be bundled into 0.3.1

**Comment by Mahgoobi** [comment]
Great.
